plugin-management-cli/src/main/java/io/jenkins/tools/pluginmanager/cli/CliOptions.java

```diff
@@ -169,13 +180,28 @@ private Plugin parsePluginLine(String pluginLine) {
         String pluginName = pluginInfo[0];
         String pluginVersion = "latest";
         String pluginUrl = null;
-        if (pluginInfo.length >= 2) {
-            pluginVersion = pluginInfo[1];
-        }
-        if (pluginInfo.length == 3) {
-            pluginUrl = pluginInfo[2];
+
+        //will default to will default to "http,https,ftp" being valid
+        UrlValidator urlValidator = new UrlValidator();
+
+        if (pluginInfo.length == 2) {
+            if (urlValidator.isValid(pluginInfo[1])) {
+                pluginUrl = pluginInfo[1];
+            }
+            else {
+                pluginVersion = pluginInfo[1];
+            }
         }
 
+        if (pluginInfo.length >= 3) {
+            pluginVersion = pluginInfo[1];
+            if (urlValidator.isValid(pluginInfo[2])) {
+                pluginUrl = pluginInfo[2];
+            }
+            else {
+                System.out.println("Invalid URL entered, will ignore");
+            }
+        }
         return new Plugin(pluginName, pluginVersion, pluginUrl);
```

Comment:
...leading to this as the final `Collect` statement.  Just an alternative way to go through this method with really the benefit coming from not needing the InputStream dance at the top.